src/hooks/Hooks.js

```diff
@@ -0,0 +1,54 @@
+import { useEffect, useState } from 'react'
+import AsyncStorage from '@react-native-async-storage/async-storage'
+
+import { getUser } from '../BaseApi'
+
+export function useUser(userId) {
+  const [user, setUser] = useState(null)
+  let isLoading = false
+
+  useEffect(() => {
+    const fetchUser = async () => {
+      try {
+        const token = await AsyncStorage.getItem('authToken')
+        const response = await getUser(userId, token)
+        if (response) {
+          if(response.statusCode === 200 && !isLoading) {
+            setUser(response.data)
+          }
+        }
+      } catch(error) {
+        error => console.warn(error)
+      }
+    }
+    fetchUser()
+    return () => isLoading = true
```

Comment:
good question, the idea was to prevent it to set the user if the view/container was re-rendered, e.g. 
1. you open the profile view of a contact and the fetch is running
2. the view re-renders for some reason (or the userId changes) and another fetch is started
3.  now if the first fetch finishes and then the second it's fine but if it's the other way round (because the first fetch had flaky internet or something) it would show the wrong user I guess ? So setting `isLoading = true` on unmount prevents that. does that make sense?